### ISSUE TITLE
RTL incorrectly truncated labels for full words mode

### ISSUE THREAD

**Issue by czogran** [amcharts/amcharts4#4214]
**Bug description**
In the right-to-left (RTL) mode, full word strings are being reversed, leading to different text results for different scenarios involving word length and maxWidth (as can be seen in the provided StackBlitz reproduction for "Czech Republic" and "The Netherlands"). This behavior is unexpected, as RTL translations should be provided in the correct word order.

Therefore, the expected behavior is that only truncation should occur, and the order of the words in the string should not be reversed in the RTL mode.

For TypeScript/Angular/React use:
[stack-blitz-example-to-reproduce](https://stackblitz.com/edit/amcharts4-mzk4gh?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.module.ts,src%2Fapp%2Fapp.component.css)

**Additional context**
Cause: 
[Utils.ts]( https://github.com/amcharts/amcharts4/blob/master/src/.internal/core/utils/Utils.ts)
function: splitTextByCharCount
fragment:
```			// Check word length
			if ((wordLength > maxChars) && (fullWords !== true || fullWordFallback != false)) {
			//if (wordLength > maxChars) {
				// A single word is longer than allowed symbol count
				// Break it up
				if (rtl) {
					word = reverseString(word);
				}
```

**Comment by martynasma** [comment]
Our text truncation algorithm does not analyze actual content to determine fuzzy logic for truncation based whether strings contain RTL, or LTR symbols, or perhaps mixed content.

In this case I can only suggest either disabling RTL, or truncation, or using your own custom logic for truncation by creating an adapter on those labels:

https://www.amcharts.com/docs/v4/concepts/adapters/

**Comment by czogran** [comment]
Thank you for your response!
I will do my best to use an adapter.

However, it is important for us to support RTL, and it would be ideal to have a universal solution. I would suggest removing the string reversal in your algorithm because the current implementation yields inconsistent results.
Furthermore, I believe that RTL is a specific way of reading, so reversing the string does not necessarily ensure correct translation.

**Comment by martynasma** [comment]
Your example uses English text with RTL settings which is not a real-world use. Can you confirm that this is still not right with actual RTL lanaguages? Can you provide example?

**Comment by czogran** [comment]
Of course, I reworked provided example to use Arabic in three first labels.
I have also found example 
https://svelte.dev/repl/7db6b074292842468eeda380d16106da?version=3.48.0

**Comment by martynasma** [comment]
Thanks.

Does this look right?
![image](https://user-images.githubusercontent.com/1458520/224292997-bd11f371-c803-4a4a-b0ef-8202294b128d.png)

If it does, we'll roll in the change into next version.

**Comment by czogran** [comment]
Thank you a lot!
I have not come up with better solution. I agree that implementing fuzzy logic is too complicated. So this compromise is OK for me :)

**Comment by martynasma** [comment]
I'm not entirely sure. Is the RTL text in my previous screenshot truncated correctly?

I need help from someone who can read it, because I can't :)

**Comment by czogran** [comment]
I asked the person who knows Arabic. She said that in Arabic text is truncated correctly.

**Comment by martynasma** [comment]
Fixed in 4.10.35.

## [4.10.35] - 2023-03-21

### Fixed
- RTL text was not being truncated correctly.
- Removed chart producing an error on longpress on touch devices in some cases.
- In some cases, when a series was removed from a Serial chart, it was not disposing one of the listeners and could remain in the memory.
